### PR TITLE
[eslint] Add es2022 nodes from estree

### PR DESCRIPTION
**Pull request by esdmr**
These nodes specifically:

- `PrivateIdentifier`: `#prop`
- `PropertyDefinition`: `prop = value;`
- `StaticBlock`: `static { … }`

Types for these nodes are already in estree. This PR just adds them to eslint.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - [ESLint v8.0.0 release notes](https://eslint.org/blog/2021/10/eslint-v8.0.0-released/)
  - [ESTree es2022 docs](https://github.com/estree/estree/blob/master/es2022.md)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by bradzacher — APPROVED** [review]

**Review by gabritto — APPROVED** [review]

**Comment by esdmr** [comment]
Ready to merge

**Comment by janhommes** [comment]
@esdmr this broke our builds:
image

Not sure atm how to fix it, as we use the angular-cli and its webpack version where an old @types/estree:0.0.50 is used. We can also not lock it, as it is the customer scaffolding. Any hint how we could avoid this?

**Comment by esdmr** [comment]
@janhommes I thought `@types/eslint` has a dependency on `@types/estree`. <sup>(Edit: I checked and it does. Is this possibly an issue with npm deduplicating?)</sup> Can you run a `npm ls --depth Infinity @types/estree` and see what depends on the old version of `@types/estree`?

**Comment by esdmr** [comment]
Wait... `@types/eslint` depends on *any* version of `@types/estree`. This is wrong!

```json
    "dependencies": {
        "@types/estree": "*",
        "@types/json-schema": "*"
    },
```

**Comment by esdmr** [comment]
I cannot action on this before an owner/maintainer can tell me if it is okay to update the `package.json` as I said in my previous comment. It seems like this is a general issue throughout DT in my quick search through the repo.

Regardless, this issue can be annoying, so here is every workaround I can think of for now. If you are affected by it, pick one (and preferably only one).

- Disable type checking of `.d.ts` files for now.
  - You can do this by setting `"skipLibCheck": true` in your `tsconfig.json` or `jsconfig.json`.
  - You lose on some type checking (obviously) and may see some `any`s around.
- Override `@types/estree` to at least `v0.0.51`.
  - You can do this in NPM via `overrides` in `package.json`:
  
    ```json
	"overrides": {
		"@types/estree@<v0.0.51": "v0.0.51"
	}
	```

  - You can also do this via a `shrinkwrap` on older NPM versions.
- Override `@types/eslint` to depend on `@types/estree@>=v0.0.51`.
  - You can do this in NPM via `overrides` in `package.json`:
  
    ```json
	"overrides": {
		"@types/eslint": {
			"@types/estree": "v0.0.51"
		}
	}
	```

  - You can also do this via a `shrinkwrap` on older NPM versions.
- Depend on `@types/estree` yourself.
  - You can do this via `npm install @types/estree@1.0.x && npm dedupe`.
  - **Not guaranteed to work.**
- Patch `@types/eslint` to the correct version of `@types/estree`.
  - Tedious. No automatic upgrades for now.

(CC: @janhommes @skyhi-github — In case you are blocked by this issue, you can try one of these workarounds.)

**Comment by janhommes** [comment]
Thanks, @esdmr for the **awesome** help. We went with the `skipLibCheck` option for now and it works. We only have the issue with Angular 12 and the latest versions don't have the issue.

Do you think a possible solution would be to update @types/eslint to not reference 9 anymore?

**Comment by esdmr** [comment]
@janhommes I can make a PR fixing this, but I am not sure if it would be accepted. (I can *yolo* it, but that feels improper.)

**Pull request merged**
